src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -137,6 +157,16 @@ void initializeReference() {
      */
     void initializeReads() {
         if (! readArguments.getReadFiles().isEmpty()) {
+            // Prefetcher is useful for cloud files because of the latencies involved.
+            Function<SeekableByteChannel, SeekableByteChannel> wrapper = Function.identity();
+            Function<SeekableByteChannel, SeekableByteChannel> indexWrapper = Function.identity();
+
+            if (cloudPrefetchBuffer > 0) {
+                wrapper = is -> GATKTool.addPrefetcher(cloudPrefetchBuffer, is);
+            }
+            if (cloudIndexPrefetchBuffer > 0) {
+                indexWrapper = is -> GATKTool.addPrefetcher(cloudIndexPrefetchBuffer, is);
+            }
```

Comment:
Yes, please move it all down into `ReadsDataSource`, and have the `ReadsDataSource` constructors take the buffer sizes as ints. There are clients in gatk-protected that use `ReadsDataSource` directly and would benefit, and in general we want to encapsulate these sort of low-level details in the data source classes as much as possible, and not bleed into the tool hierarchy.